### ISSUE TITLE
Class fn ser/de failure in Scala 2.12.3

### ISSUE THREAD

**Issue by nevillelyh** [spotify/scio#744]
`ClosureTest` fails in my WIP branch https://github.com/spotify/scio/tree/neville/scala-2.12
The class in question looks like `com.spotify.scio.values.ClosureTest$$Lambda$6035/545587540`, i.e. java 8 lambdas generated by the compiler.

```
[info] - should support class fn *** FAILED ***
[info]   java.io.IOException: Class not found
[info]   at org.objectweb.asm.ClassReader.a(Unknown Source)
[info]   at org.objectweb.asm.ClassReader.<init>(Unknown Source)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.classReader(ClosureCleaner.
scala:153)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.innerClasses(ClosureCleaner
.scala:160)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.getAccessedFields(ClosureCl
eaner.scala:176)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.storeAccessedFields(Closure
Cleaner.scala:183)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.cleanOuter(ClosureCleaner.s
cala:110)
[info]   at com.spotify.scio.util.ClosureCleaner.clean(ClosureCleaner.scala:59)
[info]   at com.spotify.scio.util.ClosureCleaner.clean$(ClosureCleaner.scala:58)
[info]   at com.spotify.scio.util.TransitiveClosureCleaner.clean(ClosureCleaner.scala:
104)
[info]   ...
```

**Comment by nevillelyh** [comment]
We're working around this with `-Ydelambdafy:inline` flag for now but it's probably required in user code as well.

**Comment by nevillelyh** [comment]
Related: https://issues.apache.org/jira/browse/SPARK-14540

**Comment by justjoheinz** [comment]
`scalacOptions ++= Seq("-Ydelambdafy:inline")` confirmed as a fix for the bug on scala 2.12.4

**Comment by nevillelyh** [comment]
Haven't hit this in a while, gonna close for now.
